dfd-wrapper/src/test/java/com/revolut/rxdata/dfd_wrapper/DataFlowDelegateTest.kt

```diff
@@ -0,0 +1,1027 @@
+package com.revolut.rxdata.dfd_wrapper
+
+import com.revolut.data.model.Data
+import com.revolut.flowdata.extensions.extractContent
+import com.revolut.rxdata.dfd_wrapper.DataFlowDelegateTest.SuspendingFrom
+import com.revolut.rxdata.dod.DodGlobal
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.TestScheduler
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.atLeastOnce
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.doSuspendableAnswer
+import org.mockito.kotlin.doThrow
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.only
+import org.mockito.kotlin.stub
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoMoreInteractions
+import org.mockito.kotlin.whenever
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+typealias Params = Int
+typealias Domain = String
+
+/*
+ * Copyright (C) 2023 Revolut
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+@ExperimentalCoroutinesApi
+class DataFlowDelegateTest {
```

Comment:
I think it shouldn't retest all the logic of the upstream DoD, only what's relevant to the exact purpose of the class - adapting Rx api to Coroutines API

Reply:
I agree, but it's problematic to test adapting logic in isolation, since we don't expose (and allow mocking) underlying DOD. Should we? Or would you suggest to just have basic functionality tests, like for original DFD?

Reply:
I think basic functionality tests in similar manner like with original DFD should be fine here

Reply:
Done 😁